cpp/src/gandiva/function_registry_string.cc

```diff
@@ -88,6 +88,23 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      "gdv_fn_initcap_utf8",
                      NativeFunction::kNeedsContext | NativeFunction::kCanReturnErrors),
 
+      NativeFunction("elt", {}, DataTypeVector{int32(), utf8(), utf8()}, utf8(),
+                     kResultNullIfNull, "gdv_fn_elt_int32_utf8_utf8",
+                     NativeFunction::kNeedsContext),
+
+      NativeFunction("elt", {}, DataTypeVector{int32(), utf8(), utf8(), utf8()}, utf8(),
+                     kResultNullIfNull, "gdv_fn_elt_int32_utf8_utf8_utf8",
```

Comment:
I dont think this can be a NULL_IF_NULL function since the output can be null if the position is invalid. This would have to be a NullInternal function

Reply:
Ok, I've looked to see an example of NullInternal, and it makes sense now.